### ISSUE TITLE
Investigate on if k8s should still use dockershim

### ISSUE THREAD

**Issue by gthao313** [bottlerocket-os/bottlerocket#3703]
The dockershim had been deprecated few years ago but we still use it. We should investigate on if we should still use it or remove it.  https://kubernetes.io/blog/2020/12/02/dockershim-faq/

**Comment by stockholmux** [comment]
Wouldn't this be a potential breaking change?

**Comment by arnaldo2792** [comment]
We have this change that started the cleanup:

https://github.com/bottlerocket-os/bottlerocket/pull/3135

In newer variants, we should investigate if we can stop symlinking the containerd socket.

**Comment by bcressey** [comment]
@gthao313 what do you mean "we still use it"? We provide a symlink from `dockershim.sock` to `containerd.sock` but Bottlerocket itself doesn't depend on `dockershim.sock` functionality.

**Comment by stockholmux** [comment]
I, personally, don't understand the consequence of the symlink staying. It's near zero cost and removing it might break someone. (Feel free to tell me I'm wrong because of XYZ!)

**Comment by bcressey** [comment]
I agree with @stockholmux - seems fine for the symlink to stick around for another five years.
